pkg/interactive/interactive_client.go

```diff
@@ -571,6 +575,10 @@ func (c *InteractiveClient) shouldExecute(line string, namedQuery bool) bool {
 		// statement has terminating ';'
 		return true
 	}
+	if !cmdconfig.Viper().GetBool(constants.ArgAutoComplete) {
```

Comment:
The `shouldExecute` function is called from the `executor` which is called only when the query needs execution. `Execution` happens later. 

I think a variant of this should go in at `InteractiveClient.queryCompleter`.

Reply:
Does this need to be there? The `shouldExecute` does not get called till the actual Execution time.

Autocomplete setting should not influence the decision to execute. Can you please remove this?